flink-libraries/flink-table/src/main/scala/org/apache/flink/table/descriptors/SchematicDescriptor.scala

```diff
@@ -19,14 +19,17 @@
 package org.apache.flink.table.descriptors
 
 /**
-  * Common class for all descriptors describing a table sink.
+  * A trait for descriptors that allow to define a format and schema.
   */
-abstract class TableSinkDescriptor extends TableDescriptor {
+trait SchematicDescriptor extends Descriptor {
```

Comment:
are you using it anywhere as interface? what does extracting it to separate interface give us? Maybe drop it?

Reply:
`SchematicDescriptor` is used for `ExternalCatalogTable`, `StreamTableDescriptor`, and `BatchTableDescriptor`. If we add a new level next to `connector`, `format` (which may happens in the future), we would immediately get a compile error there.